### PR TITLE
Theme GTK3/4 window controls to match COSMIC theme

### PR DESCRIPTION
**Pull request by nabrious0**
#### This PR relies on the COSMIC GTK3/4 theming feature and extends it to also theme window control buttons 
_(This covers colors, hover background color, window active/inactive state colors, transitions (animations) and more...)_ 

**Examples below ( ↓ )**

**Eddy - GTK3 (Light/Dark)**
image
image

**Flatseal - GTK4|libadwaita (Light/Dark)**
image
image

**Main Menu - GTK4|libadwaita (Light/Dark)**
image
image

**Nautilus (GNOME Files) - GTK3 (Light/Dark)**
image
image

### CODE REVIEW

**Comment by mmstick** [comment]
Is it possible to make all icons in the header bar match the accent color?

**Comment by nabrious0** [comment]
> Is it possible to make all icons in the header bar match the accent color?

Haven't looked into it but I'm 99.99% sure it's possible, I'll check and report back.

**Comment by nabrious0** [comment]
> Is it possible to make all icons in the header bar match the accent color?

image

Yup, it is.

**Comment by nabrious0** [comment]
**Here are updated screenshots to include the colored icons:**

**Main Menu - GTK4|libadwaita**
image
image

**Flatseal - GTK4|libadwaita**
image
image

**Nautilus (GNOME Files) - GTK3**
image
image

**Comment by mmstick** [comment]
I see that some icons still aren't being colored correctly, but that's closer to our designs than before. All icons should have the accent color whenever a window is focused, and then an inactive grey color when the window is unfocused. I'm not sure if it's possible to change the color based on window focus status in GTK.

**Comment by nabrious0** [comment]
> I see that some icons still aren't being colored correctly, but that's closer to our designs than before. All icons in the header bar should have the accent color whenever a window is focused, and then an inactive grey color when the window is unfocused. I'm not sure if it's possible to change the color based on window focus status in GTK.

It is possible, on libadwaita that is `.selector**:backdrop** {}` and I already do that for libadwaita on GTK3 I have yet to find the selector.

**Comment by nabrious0** [comment]
Colors even more icons according to accent color, and makes them all gray when a window is inactive (currently exclusive to libadwaita)

image
image

**Comment by git-f0x** [comment]
Just a note: afaik COSMIC apps use either 8px (Round, Slightly Round) or 2px (Square) window corner radius, at least from looking at the config in Settings.

**Comment by nabrious0** [comment]
> Just a note: afaik COSMIC apps use either 8px (Round, Slightly Round) or 2px (Square) window corner radius, at least from looking at the config in Settings.

Huh interesting, cause after setting it to 6px and putting a COSMIC app and libadwaita app on top of each other, they seemed to perfectly match, I'll double check.

File: cosmic-theme/src/output/gtk4_output.rs [hunk 1]
```diff
@@ -73,6 +73,12 @@ impl Theme {
             Rgba::new(0.0, 0.0, 0.0, 0.08)
         });
 
+        let window_control_hover_bg = to_rgba(if self.is_dark {
+            Rgba::new(255.0, 255.0, 255.0, 0.12)
+        } else {
+            Rgba::new(0.0, 0.0, 0.0, 0.12)
+        });
+
```
Comment (wash2) [inline cosmic-theme/src/output/gtk4_output.rs:81]:
```suggestion
        let corner_radius = self.corner_radii.radius_s[0];
```

File: cosmic-theme/src/output/gtk4_output.rs [hunk 2]
```diff
@@ -113,6 +119,74 @@ impl Theme {
...
+.toggle:backdrop > arrow,
+button:backdrop > image,
+button:backdrop > widget > box > image,
+stack:backdrop > box > image,
+.toggle:backdrop > box > contents > image {{
+	filter: grayscale(1) brightness(1.7);
+}}
+window {{ background: @window_bg_color }}
+window:not(.maximized) {{
+	border-radius: 6px;
```
Comment (wash2) [inline cosmic-theme/src/output/gtk4_output.rs:186]:
```suggestion
	border-radius: {corner_radius:.0}px;
```

**Review by wash2 — CHANGES_REQUESTED** [review]
This looks great! I've just suggested a couple small changes.

**Review by wash2 — APPROVED** [review]
I applied the suggestions.

**Comment by nabrious0** [comment]
Awesome! I didn't even know getting the real border radii was possible, thanks!

**Comment by jackpot51** [comment]
No, we cannot reinstate this.

**Pull request merged**